NetKet/Graph/abstract_graph.hpp

```diff
@@ -15,9 +15,27 @@
 #ifndef NETKET_ABSTRACTGRAPH_HPP
 #define NETKET_ABSTRACTGRAPH_HPP
 
+#include <array>
+#include <unordered_map>
 #include <vector>
 
+// Special hash functor for the EdgeColors unordered_map
+// Same as hash_combine from boost
+namespace std {
+struct ArrayHasher {
+  std::size_t operator()(const std::array<int, 2>& a) const {
+    std::size_t h = 0;
+
+    for (auto e : a) {
```

Comment:
Why so difficult? This is the way you'd implement a hash function for `std::array<int, N>`, but `std::array<int, 2>` is just a piece of memory 8 bytes long, and a hash is 8 bytes long, so I don't see why not simply cast one into the other?

Reply:
I see now, I was making things more difficult than necessary. I've modified your initial suggestion so the hash functor is now 
```cpp
struct ArrayHasher {
  std::size_t operator()(const std::array<int, 2>& a) const {
    return *reinterpret_cast<std::size_t const*>(a.data());
  }
};
```

Reply:
You might want to plug `noexcept` in there, but otherwise it looks good to me.

Reply:
I guess `ArrayHasher` is not specific of `Graph`, and it is better moved in the `Utils/`. Maybe a new file `hashing.hpp` or something similar?

Reply:
Sure, I'll create the `array_hasher.hpp` file in `Utils/` with the noexcept.